ui/src/preview/preview-table-item.tsx

```diff
@@ -62,33 +68,135 @@ export const PreviewTableItem: React.FC<PreviewTableItemProps> = ({
+                displayedPublicationStage,
+                moveTargetStage,
+            ),
+        ),
+        name: t('publish.move-stage-changes', {
+            amount: stagePublicationAssetAmount,
+        }),
+    };
+
+    const menuOptionMovePublicationGroupStage = {
+        onSelect: menuAction(() => {
+            if (tableEntry.publicationGroup) {
+                previewOperations.setPublicationStage.forPublicationGroup(
+                    tableEntry.publicationGroup,
+                    moveTargetStage,
+                );
+            }
+        }),
+        name: t('publish.move-publication-group', {
+            amount: publicationGroupAssetAmount,
+        }),
+    };
+
+    const menuOptionRevertSingleChange = {
+        name: t('publish.revert-change'),
+        onSelect: menuAction(() =>
+            previewOperations.revert.changesWithDependencies(
+                tableEntryAsPublishRequestIds,
+                tableEntryAsRevertRequestSource,
+            ),
+        ),
+    };
+
+    const menuOptionRevertStageChanges = {
+        onSelect: menuAction(() => {
+            previewOperations.revert.stageChanges(
+                displayedPublicationStage,
+                tableEntryAsRevertRequestSource,
+            );
+        }),
+        name: t('publish.revert-stage-changes', {
+            amount: stagePublicationAssetAmount,
+        }),
+    };
+
+    const menuOptionPublicationGroupRevert = {
+        onSelect: menuAction(() => {
+            if (tableEntry.publicationGroup) {
+                previewOperations.revert.publicationGroup(
+                    tableEntry.publicationGroup,
+                    tableEntryAsRevertRequestSource,
+                );
+            }
+        }),
+        name: t('publish.revert-publication-group', {
+            amount: publicationGroupAssetAmount,
+        }),
+    };
+
+    const menuOptionShowOnMap = {
+        disabled: !tableEntry.boundingBox,
+        onSelect: menuAction(() => {
+            tableEntry.boundingBox && onShowOnMap(tableEntry.boundingBox);
+        }),
+        name: t('publish.show-on-map'),
+    };
+
     const menuOptions = [
-        {
-            disabled: !boundingBox,
-            onSelect: () => {
-                boundingBox && onShowOnMap(boundingBox);
-                setActionMenuVisible(false);
-            },
-            name: t('publish.show-on-map'),
-        },
-        {
-            name: t('publish.revert-change'),
-            onSelect: () => {
-                onRevert();
-                setActionMenuVisible(false);
-            },
-        },
+        ...(tableEntry.publicationGroup ? [menuOptionMovePublicationGroupStage] : []),
+        menuOptionMoveStageChanges,
+        menuOptionShowOnMap,
+        ...(!tableEntry.publicationGroup ? [menuOptionRevertSingleChange] : []),
+        ...(tableEntry.publicationGroup ? [menuOptionPublicationGroupRevert] : []),
+        menuOptionRevertStageChanges,
```

Comment:
Tulkitsin suht järkeväksi syntaksiksi määritellä tuo näytettävä menu per rivikonteksti. Tosin menee kyllä heti epäluettavammaksi, jos näitä ehtoja tulee lisää per tietty menu-item.

Reply:
Tämä saattaisi mennä luettavampaan muotoon mikäli nuo ehdolliset ottaisi omiin taulukkovälimuuttujiinsa ja sitten tekisi täällä vaan `...välimuuttuja`

Reply:
Tämä saattaisi mennä, mutta sitten taas jokaiseen menuOptioniin tulee aika ikävää syntaksia jos ne suoraan wrappaa arrayihin.

🤔 Voisi tämän tietysti funktioida, esim. `...conditionalMenuOption(condition, menuOption)` (taidan tehdäkin tällaisen)

Reply:
Tämä conditionalMenuOption lisäilty.